utils/service_discovery/sd_docker_backend.py

```diff
@@ -225,43 +225,37 @@ def _extract_ip_from_networks(self, ip_dict, tpl_var):
         if len(tpl_parts) < 2:
             log.debug("No key was passed for template variable %s." % tpl_var)
             return self._get_fallback_ip(ip_dict)
-        else:
-            res = ip_dict.get(tpl_parts[-1])
-            if res is None:
-                log.warning("The key passed for template variable %s was not found." % tpl_var)
-                return self._get_fallback_ip(ip_dict)
-            else:
-                return res
+
+        res = ip_dict.get(tpl_parts[-1])
+        if res is None:
+            log.warning("The key passed for template variable %s was not found." % tpl_var)
+            return self._get_fallback_ip(ip_dict)
+        return res
 
     def _get_fallback_ip(self, ip_dict):
         """try to pick the bridge key, falls back to the value of the last key"""
         if 'bridge' in ip_dict:
             log.debug("Using the bridge network.")
             return ip_dict['bridge']
-        else:
-            last_key = sorted(ip_dict.iterkeys())[-1]
-            log.debug("Trying with the last (sorted) network: '%s'." % last_key)
-            return ip_dict[last_key]
+
+        last_key = sorted(ip_dict.iterkeys())[-1]
+        log.debug("Trying with the last (sorted) network: '%s'." % last_key)
+        return ip_dict[last_key]
 
     def _get_port(self, state, c_id, tpl_var):
         """Extract a port from a container_inspect or the k8s API given a template variable."""
         container_inspect = state.inspect_container(c_id)
-
+        ports = []
         try:
-            ports = map(lambda x: x.split('/')[0], container_inspect['NetworkSettings']['Ports'].keys())
-            if len(ports) == 0: # There might be a key Port in NetworkSettings but no ports so we raise IndexError to check in ExposedPorts
+            ports = [x.split('/')[0] for x in container_inspect['NetworkSettings']['Ports'].keys()]
+            if len(ports) == 0:
                 raise IndexError
         except (IndexError, KeyError, AttributeError):
-            # try to get ports from the docker API. Works if the image has an EXPOSE instruction
-            ports = map(lambda x: x.split('/')[0], container_inspect['Config'].get('ExposedPorts', {}).keys())
```

Comment:
That line might get AD working for one customer with an exotic version. Instead of removing it, I'd go with
```
if Platform.is_k8s():
    kube_way
else:
   exposedports


Just as a precaution

Reply:
that's a good idea

Reply:
@xvello fixed